From 251f2266bf24b1b396f59eef60d0acf41fdd02e4 Mon Sep 17 00:00:00 2001
From: Peter S. Mazinger <ps.m@gmx.net>
Date: Fri, 18 Mar 2011 14:37:35 +0000
Subject: lutimes.c, stubs.c: fix compiling lutimes, if __NR_utimensat is not defined

while there, provide stubs for functions depending on utimensat syscall.

Reported-by: Sedat Dilek <sedat.dilek@googlemail.com>
Signed-off-by: Peter S. Mazinger <ps.m@gmx.net>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
diff --git a/libc/sysdeps/linux/common/lutimes.c b/libc/sysdeps/linux/common/lutimes.c
index 0b4a8ea..e01d40e 100644
--- a/libc/sysdeps/linux/common/lutimes.c
+++ b/libc/sysdeps/linux/common/lutimes.c
@@ -12,7 +12,7 @@
 
 #ifdef __NR_lutimes
 _syscall2(int, lutimes, const char *, file, const struct timeval *, tvp)
-#else
+#elif defined __NR_utimensat
 #include <sys/time.h>
 #include <fcntl.h>
 
diff --git a/libc/sysdeps/linux/common/stubs.c b/libc/sysdeps/linux/common/stubs.c
index 8688e98..655c640 100644
--- a/libc/sysdeps/linux/common/stubs.c
+++ b/libc/sysdeps/linux/common/stubs.c
@@ -172,6 +172,14 @@ make_stub(umount)
 make_stub(umount2)
 #endif
 
+#ifndef __NR_utimensat
+make_stub(futimens)
+make_stub(utimensat)
+# ifndef __NR_lutimes
+make_stub(lutimes)
+# endif
+#endif
+
 #ifndef __NR_vmsplice
 make_stub(vmsplice)
 #endif
--
cgit v0.8.2.1
